src/association/mod.rs

```diff
@@ -80,12 +80,14 @@ impl AssociationParameters {
 
         for (i, record) in records.iter().enumerate() {
             if let Some(record) = record.as_ref() {
-                assoc_comp.push(i);
-                sigma_assoc.push(sigma[i]);
-                kappa_ab.push(record.kappa_ab);
-                epsilon_k_ab.push(record.epsilon_k_ab);
-                na.push(record.na.unwrap_or(1.0));
-                nb.push(record.nb.unwrap_or(1.0));
+                if record.kappa_ab > 0.0 && record.epsilon_k_ab > 0.0 {
```

Comment:
if purely numerical input is used (e.g., from a large dataset) this speeds up the computation.